src/NodeParser/StringTemplateLiteralNodeParser.ts

```diff
@@ -0,0 +1,73 @@
+import ts from "typescript";
+import { UnknownTypeError } from "../Error/UnknownTypeError";
+import { Context, NodeParser } from "../NodeParser";
+import { SubNodeParser } from "../SubNodeParser";
+import { AliasType } from "../Type/AliasType";
+import { BaseType } from "../Type/BaseType";
+import { LiteralType } from "../Type/LiteralType";
+import { UnionType } from "../Type/UnionType";
+
+export class StringTemplateLiteralNodeParser implements SubNodeParser {
+    public constructor(protected childNodeParser: NodeParser) {}
+
+    public supportsNode(node: ts.NoSubstitutionTemplateLiteral | ts.TemplateLiteralTypeNode): boolean {
+        return (
+            node.kind === ts.SyntaxKind.NoSubstitutionTemplateLiteral || node.kind === ts.SyntaxKind.TemplateLiteralType
+        );
+    }
+    public createType(node: ts.NoSubstitutionTemplateLiteral | ts.TemplateLiteralTypeNode, context: Context): BaseType {
+        if (node.kind === ts.SyntaxKind.NoSubstitutionTemplateLiteral) {
+            return new LiteralType(node.text);
+        }
+        const prefix = node.head.text;
+        const matrix: string[][] = [[prefix]];
+
+        for (const span of node.templateSpans) {
+            const suffix = span.literal.text;
+
+            const type = this.childNodeParser.createType(span.type, context);
+            const literals = [...extractLiterals(type)];
+
+            matrix.push(literals.map((value) => value + suffix));
+        }
+
+        const expandedLiterals = expand(matrix);
+
+        const expandedTypes = expandedLiterals.map((literal) => new LiteralType(literal));
+
+        if (expandedTypes.length === 1) {
+            return expandedTypes[0];
+        }
+
+        return new UnionType(expandedTypes);
+    }
+}
+
+function expand(matrix: string[][]): string[] {
+    if (!matrix.length) return [];
+    if (matrix.length === 1) return matrix[0];
```

Comment:
Let's convert that into a switch case statement over the length of the matrix.

Reply:
In this case `matrix.length`, so I removed the `if (!matrix.length) return [];` line.